src/TabSwipe/widget/TabSwipe.ts

```diff
@@ -0,0 +1,173 @@
+import * as dojoDeclare from "dojo/_base/declare";
+import * as WidgetBase from "mxui/widget/_WidgetBase";
+import * as aspect from "dojo/aspect";
+import * as domConstruct from "dojo/dom-construct";
+import * as registry from "dijit/registry";
+
+import { SwipeHandler } from "./SwipeHandler";
+import "./ui/TabSwipe.scss";
+
+export interface TabContainer extends mxui.widget._WidgetBase {
+    declaredClass: "mxui.widget.TabContainer";
+    showTab: (tab: TabPane) => void;
+    focusIndex: number;
+    onShowTab: (callback: () => void ) => void;
+    onHideTab: (callback: () => void ) => void;
+    tabSwipeId: string;
+    validator: null;
+    _active: TabPane;
+    _tabList: HTMLElement;
+    _tabPanes: TabPane[];
+    _tabContent: TabContent;
+    _clickHandler: null;
+}
+
+export interface TabContent extends HTMLElement {
+    declaredClass: "mxui.widget.TabContent";
+}
+
+export interface TabPane extends mxui.widget._WidgetBase {
+    _hidden: boolean;
+    _parsed: boolean;
+    _visible: boolean;
+    container: TabContainer;
+    showTab: (tab: TabPane) => void;
+    show: () => void;
+    hideTab: () => void;
+    index: number;
+    visibilityIndex: number;
+}
+
+class TabSwipe extends WidgetBase {
+    targetName: string;
+    tabStyle: "tabs"| "indicators";
+    lazyLoad: boolean;
+
+    private targetWidget?: TabContainer;
+    private targetNode: HTMLElement;
+    private swipeHandler: SwipeHandler;
+    private tabContentClass: string;
+    private swipeClass: string;
+    private targetWidgetType: string;
+
+    postCreate() {
+        this.tabContentClass = ".mx-tabcontainer-content";
+        this.swipeClass = "widget-tab-swipe";
+        this.targetWidgetType = "mxui.widget.TabContainer";
+        this.targetNode = this.findTargetNode(this.targetName);
+        this.targetWidget = this.targetNode ? registry.byNode(this.targetNode) : null;
+
+        if (!this.targetWidget) {
+            this.showError(`Tab swipe configuration error: unable to find a target with the name ${this.targetName}`);
```

Comment:
If the tabs swipe widget should be place below the tab, it should be mentioned in the error, as it could be that the user places it  in the wrong place.
`, make sure that the widget is placed directly blow a tab container`